test/fixtures/rules/std_rule_cases/L007.yml

```diff
@@ -98,20 +136,85 @@ passes_on_after_override:
 fails_on_before_override_with_comment_order:
   fail_str: |
     select
-        a
-        -- comment1!
+        a -- comment1!
         -- comment2!
+        -- comment3!
         + b
     from foo
   fix_str: |
     select
-        a +
-        -- comment1!
+        a + -- comment1!
         -- comment2!
+        -- comment3!
         b
     from foo
   configs: *operator_before
 
+fails_on_after_override_with_comment_order:
+  fail_str: |
+    select
+        a + -- comment1!
+        -- comment2!
+        -- comment3!
+        b
+    from foo
+  fix_str: |
+    select
+        a -- comment1!
+        -- comment2!
+        -- comment3!
+        + b
+    from foo
+  configs: *operator_after
+
+# Fix the different variations of problematic comments in a leading case.
+fails_leading_with_comments:
+  fail_str: |
+    SELECT
+      1 +
+      /* foo */ 2,
+      1 + -- foo
+      2,
+      1 /* foo */ +
+      2,
+      1 /* foo */ + -- foo
+      /* foo */ 2
+  fix_str: |
+    SELECT
+      1
+      /* foo */ + 2,
+      1 -- foo
+      + 2,
+      1 /* foo */
+      + 2,
+      1 /* foo */ -- foo
```

Comment:
These tests with interspersed comments remind me: sometimes I think it'd be interesting to have a "fuzz" tester that takes our existing test cases and throws randomly placed comments into them to see if something blows up. Think that may be useful?

Reply:
I agree - and I'm thinking about doing something like that when we get to the full blown reflow command. That way we can randomly comment and space sql files and then check that the reflow command is deterministic. I'll add a task on the master issue to do that.